### PR TITLE
Update duckdb to the newest version

### PR DESCRIPTION
**Pull request by Taepper**
## PR Checklist

- [ ] All necessary documentation has been adapted or there is an issue to do so.
- [ ] The implemented feature is covered by an appropriate test.

### CODE REVIEW

File: endToEndTests/test/queries/nOf_2of3_details.json
Comment (fengelniederhammer) [inline endToEndTests/test/queries/nOf_2of3_details.json:1]: This test fails for me locally...
```
  Error [AssertionError]: Actual result is:
  {"age":58,"country":"Switzerland","date":"2021-04-28","division":"Basel-Stadt","gisaid_epi_isl":"EPI_ISL_2019235","pango_lineage":"B.1.1.7","qc_value":0.9,"region":"Europe","test_boolean_column":false,"unsorted_date":"2021-01-22"}
  {"age":54,"country":"Switzerland","date":"2021-03-19","division":"Solothurn","gisaid_epi_isl":"EPI_ISL_1597932","pango_lineage":"B.1.1.7","qc_value":0.94,"region":"Europe","test_boolean_column":true,"unsorted_date":"2021-02-10"}
  {"age":50,"country":"Switzerland","date":"2020-11-13","division":"Solothurn","gisaid_epi_isl":"EPI_ISL_1005148","pango_lineage":"B.1.221","qc_value":0.92,"region":"Europe","test_boolean_column":null,"unsorted_date":"2020-12-17"}
  {"age":50,"country":"Switzerland","date":"2021-02-23","division":"Solothurn","gisaid_epi_isl":"EPI_ISL_1195052","pango_lineage":"B.1.1.7","qc_value":0.95,"region":"Europe","test_boolean_column":null,"unsorted_date":"2021-07-04"}

```

**Review by fengelniederhammer — APPROVED** [review]

**Comment by fengelniederhammer** [comment]
And maybe you want to squash your commits? Every `fix` commit will end up in the changelog with its commit message.

File: endToEndTests/test/queries/nOf_2of3_details.json
Comment (Taepper) [reply endToEndTests/test/queries/nOf_2of3_details.json:1]: Then I would wait for christians new streaming implementation, which might make the samples determinstic across platforms

File: conanfile.py [hunk 1]
```diff
@@ -7,7 +7,7 @@ class SiloRecipe(ConanFile):
 
     requires = [
         "boost/1.82.0",
-        "duckdb/0.8.1",
+        "duckdb/0.10.2",
```
Comment (corneliusroemer) [inline conanfile.py:10]:
```suggestion
        "duckdb/0.10.3",
```

**Review by fengelniederhammer — APPROVED** [review]

**Pull request merged**